codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/RestJson1.kt

```diff
@@ -618,6 +618,8 @@ class RestJson1HttpDeserializerGenerator(
                 "let request = request.map(|body| #{Bytes}::from(body.bytes().unwrap().to_vec()));",
                 *codegenScope
             )
+            rust("assert_eq!(request.method().as_str(), \"${testCase.method}\");")
+            rust("assert_eq!(request.uri().to_owned(), \"${testCase.uri}\");")
```

Comment:
I'm not sure how this works...fwiw, when comparing Uris I usually generated `Uri::from_static` for the comparison.